src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -503,6 +514,121 @@ private static void createBranches(List<LfBus> lfBuses, LfNetwork lfNetwork, LfT
+     */
+    private static void addTieLineAreaBoundaries(TieLine tieLine, LfBranch lfTieLineBranch, LoadingContext loadingContext) {
+        addAreaBoundary(tieLine.getTerminal1(), lfTieLineBranch, TwoSides.TWO, loadingContext);
+        addAreaBoundary(tieLine.getTerminal2(), lfTieLineBranch, TwoSides.ONE, loadingContext);
+    }
+
+    /**
+     * Adds the active power of the terminal of the dangling line to the calculation of the Area's interchange (load convention) if it is a boundary.
+     * The equivalent injection of the dangling line lfBranch model is P2;
+     */
+    private static void addDanglingLineAreaBoundary(DanglingLine danglingLine, LfBranch lfDanglingLineBranch, LoadingContext loadingContext) {
+        addAreaBoundary(danglingLine.getTerminal(), lfDanglingLineBranch, TwoSides.TWO, loadingContext);
+    }
+
+    private static void addAreaBoundary(Terminal terminal, LfBranch branch, TwoSides side, LoadingContext loadingContext) {
+        if (loadingContext.areaTerminalMap.containsKey(terminal)) {
+            Area area = loadingContext.areaTerminalMap.get(terminal);
+            loadingContext.areaBoundaries.computeIfAbsent(area, k -> new HashSet<>()).add(new LfAreaImpl.BoundaryImpl(branch, side));
+        }
+    }
+
+    private static void createAreas(int numCC, int numSC, LoadingContext loadingContext, List<LfNetworkLoaderPostProcessor> postProcessors, LfNetworkParameters parameters, LfNetwork network) {
+        if (parameters.isAreaInterchangeControl()) {
+            loadingContext.areaBusMap.forEach((area, lfBuses) -> {
+                checkBoundariesComponent(area, numCC, numSC);
+                Set<LfArea.Boundary> boundaries = loadingContext.areaBoundaries.getOrDefault(area, new HashSet<>());
+                LfArea lfArea = LfAreaImpl.create(area, lfBuses, boundaries, network, parameters);
+                network.addArea(lfArea);
+                postProcessors.forEach(pp -> pp.onAreaAdded(area, lfArea));
+            });
+            checkBusesWithoutArea(network);
+        }
+    }
+
+    private static void checkBoundariesComponent(Area area, int numCC, int numSC) {
+        area.getAreaBoundaryStream().forEach(boundary -> {
+            Bus bus;
+            if (boundary.getTerminal().isPresent()) {
+                Terminal terminal = boundary.getTerminal().get();
+                bus = terminal.getBusBreakerView().getConnectableBus();
+            } else if (boundary.getBoundary().isPresent()) {
+                DanglingLine danglingLine = boundary.getBoundary().get().getDanglingLine();
+                bus = danglingLine.getTerminal().getBusBreakerView().getConnectableBus();
+            } else {
+                return;
+            }
+            if (bus.getConnectedComponent().getNum() != numCC || bus.getSynchronousComponent().getNum() != numSC) {
```

Comment:
caution, `bus.getConnectedComponent()` / `bus.getSynchronousComponent()` can be null, would thrown NPE here